source/1_engine/rex_engine/src/win/win_com_library.cpp

```diff
@@ -0,0 +1,102 @@
+#include "rex_engine/win/win_com_library.h"
+
+#include <coml2api.h>
+#include <objbase.h>
+#include <objidl.h>
+#include <shlguid.h>
+#include <shobjidl_core.h>
+#include <wtypes.h>
+
+thread_local card32 rex::win::ComLibrary::s_init_succeeded_count = 0;
+
+rex::win::ComLibrary::ComLibrary()
+{
+  if(s_init_succeeded_count == 0)
```

Comment:
Could this class also benefit from a ref count object?
Also is the reason why it's refcounted cuz you didn't wanna make a singleton out of it?